dask/array/_shuffle.py

```diff
@@ -79,7 +87,44 @@ def shuffle(x, indexer: list[list[int]], axis):
     return Array(graph, out_name, chunks, meta=x)
 
 
-def _shuffle(chunks, indexer, axis, in_name, out_name, token):
+def _resize_other_dimensions(x: Array, longest_group, axis, chunks):
+    assert chunks == "auto", "Only auto is supported for now"
+    chunksize_tolerance = config.get("array.shuffle.chunksize-tolerance")
+
+    if longest_group <= max(x.chunks[axis]) * chunksize_tolerance:
+        # We are staying below our threshold, so don't rechunk
+        return x
+
+    max_group = reduce(mul, map(max, x.chunks))
+    new_max_group = max_group / max(x.chunks[axis]) * longest_group
+
+    new_chunks = []
+    for i in range(len(x.chunks)):
+        if i == axis:
+            new_chunks.append(x.chunks[i])
+            continue
+
+        new_chunk = []
+        # calculate what the max chunk size in this dimension is and split every
+        # chunk that is larger than that
+        dimension_max_chunk = max(x.chunks[i]) * (
+            (max_group / new_max_group) ** (1 / (len(x.chunks) - 1))
+        )
+        for c in x.chunks[i]:
+            if c > chunksize_tolerance * dimension_max_chunk:
+                factor = c / dimension_max_chunk
+                nc = [c // math.ceil(factor)] * math.ceil(factor)
+                for ii in range(c - sum(nc)):
+                    nc[ii] += 1
+                new_chunk.extend(nc)
```

Comment:
don't follow

Reply:
Bad variable naming probably. We are building the new chunk sizes for the axis we are currently investigating.